report_generation/report_generation.py

```diff
@@ -1,49 +1,90 @@
-        ) = self._get_output_file_path(self.file_path, self.file_name)
-
-    def _get_output_file_path(self, file_path: str, file_name: str) -> tuple[str, str]:
-        file_path_frequency = f"{file_path}/frequency-reports-{DatetimeHelper.get_current_date()} {StringHelper.extract_time_from_file_name(file_name)}"
-        file_path_booking = f"{file_path}/booking-reports"
+        self.frequency_file_name, self.collection_file_name = self._map_input_files(file_path)
+
+        self.output_file_path_booking, self.output_file_path_frequency = self._get_output_file_path(file_path)
+
+    def _map_input_files(self, file_path: str) -> tuple[str, str]:
+        # TODO: Create a map of input files to their respective types so we can see which files are being processed
+        # E.g. {frequency_report: file_name, collections_report: file_name}
+        # Either use string matching or a regex to build this.
+        # Use the enum below
+        frequency_file_name = None
+        collection_file_name = None
+        
+        files = os.listdir(file_path)
+        for file in files:
+            if file.startswith("FR Report"):
+                frequency_file_name = file
+            elif file.startswith("Collection"):
+                collection_file_name = file
+
+        if frequency_file_name is None:
+            raise FileNotFoundError(f"Frequency report file not found in directory '{file_path}'")
+        if collection_file_name is None:
+            raise FileNotFoundError(f"Collection report file not found in directory '{file_path}'")
+
+        return frequency_file_name, collection_file_name
+
+    def _get_output_file_path(self, file_path: str) -> tuple[str, str]:
+        output_file_path = os.path.join(file_path, os.path.basename(file_path))
+        file_path_frequency = f"{output_file_path}/frequency-reports-{DatetimeHelper.get_current_date()} {StringHelper.extract_time_from_file_name(self.frequency_file_name)}"
+        file_path_booking = f"{output_file_path}/booking-reports"
 
         OSHelper.create_directories([file_path_booking, file_path_frequency])
         return file_path_booking, file_path_frequency
 
-    def preprocess_data(self, df: pd.DataFrame) -> pd.DataFrame:
+    def preprocess_data(self, df: pd.DataFrame, is_collection_report: bool = False) -> pd.DataFrame:
         # TODO: Ensure df.columns match the minimum expected columns (do a set comparison of expected vs. current)
-        date_columns = ["Waybill Date", "Due Date", "POD Date", "Last Event Date"]
+        if is_collection_report:
+        # Specify the date columns for collection report
+            date_columns = ["Date", "Capture Date"] 
+        else:
+            # Date columns for frequency report
+            date_columns = ["Waybill Date", "Due Date", "POD Date", "Last Event Date"]
+
         for col in date_columns:
-            df[col] = df[col].apply(DatetimeHelper.safe_to_date)
+            if col in df.columns:  # Only process if column exists
```

Comment:
This could mean that we just have one thing of Date columns and it will check. I actually like throwing an error because we need to know the date columns so we can format them. Can we remove this?